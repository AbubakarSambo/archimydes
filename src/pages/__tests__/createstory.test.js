/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import UserStory from '../createstory'

describe('Login Component', () => {
  let component
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    })
    const history = createMemoryHistory()
    const { getByTestId } = render(
      <Router history={history}>
        <UserStory />
      </Router>
    )
    component = getByTestId('create')
  })

  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
