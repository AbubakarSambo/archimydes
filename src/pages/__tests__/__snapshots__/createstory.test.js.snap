// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Should match snapshot 1`] = `
<div
  class="sc-AxjAm csqjiX"
  data-testid="create"
>
  <div>
    <div
      class="sc-AxirZ eUyPUG"
    >
      <h3>
        Create a Story
      </h3>
      <button
        class="ant-btn ant-btn-danger"
        type="button"
      >
        <span>
          Log out
        </span>
      </button>
    </div>
    <form
      class="ant-form ant-form-horizontal"
      id="basic"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_summary"
            title="Summary"
          >
            Summary
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="basic_summary"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_description"
            title="Description"
          >
            Description
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <textarea
                class="ant-input"
                id="basic_description"
                rows="5"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_type"
            title="Type"
          >
            Type
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-select ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="basic_type_list_0"
                      aria-autocomplete="list"
                      aria-controls="basic_type_list"
                      aria-haspopup="listbox"
                      aria-owns="basic_type_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="basic_type"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  />
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_complexity"
            title="Complexity"
          >
            Complexity
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-select ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="basic_complexity_list_0"
                      aria-autocomplete="list"
                      aria-controls="basic_complexity_list"
                      aria-haspopup="listbox"
                      aria-owns="basic_complexity_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="basic_complexity"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  />
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_estimatedHrs"
            title="Estimated Time(Hrs)"
          >
            Estimated Time(Hrs)
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="basic_estimatedHrs"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_cost"
            title="Cost"
          >
            Cost
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-input-number"
              >
                <div
                  class="ant-input-number-handler-wrap"
                >
                  <span
                    aria-disabled="false"
                    aria-label="Increase Value"
                    class="ant-input-number-handler ant-input-number-handler-up "
                    role="button"
                    unselectable="unselectable"
                  >
                    <span
                      aria-label="up"
                      class="anticon anticon-up ant-input-number-handler-up-inner"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="up"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    aria-disabled="false"
                    aria-label="Decrease Value"
                    class="ant-input-number-handler ant-input-number-handler-down "
                    role="button"
                    unselectable="unselectable"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-input-number-handler-down-inner"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <div
                  class="ant-input-number-input-wrap"
                >
                  <input
                    aria-valuemin="-9007199254740991"
                    autocomplete="off"
                    class="ant-input-number-input"
                    id="basic_cost"
                    min="-9007199254740991"
                    role="spinbutton"
                    step="1"
                    value="$ "
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn ant-btn-primary"
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
